crates/transaction-pool/src/traits.rs

```diff
@@ -921,13 +924,13 @@ pub trait PoolTransaction: fmt::Debug + Send + Sync + Clone {
     fn chain_id(&self) -> Option<u64>;
 }
 
-/// An extension trait that provides additional interfaces for the
-/// [`EthTransactionValidator`](crate::EthTransactionValidator).
+/// Super trait for transactions that can be converted to and from Eth transactions
 pub trait EthPoolTransaction:
     PoolTransaction<
-    Pooled = PooledTransactionsElementEcRecovered,
-    Consensus = TransactionSignedEcRecovered,
->
+        Consensus: From<TransactionSignedEcRecovered> + Into<TransactionSignedEcRecovered>,
+        Pooled: From<PooledTransactionsElementEcRecovered>
+                    + Into<PooledTransactionsElementEcRecovered>,
```

Comment:
relaxing this makes sense, until we have proper trait abstraction for primitives